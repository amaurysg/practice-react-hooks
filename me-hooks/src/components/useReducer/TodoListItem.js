import React, { useState } from "react";
import "./todoStyle.css";

const TodoListItem = ({
  todo,
  i,
  handleDelete,
  handleToogle,
  handleUpdate,
}) => {
  const [editDesc, setEditDesc] = useState(false);

  const [newDesc, setNewDesc] = useState(todo.desc);

  const changeEditDesc = (e) => {
    // alert("fila");

    console.log("chanfe");
    // setNewDesc(newDesc);
    console.log("newDesc:", newDesc);
    setEditDesc(!editDesc);
    console.log(newDesc);
  };

  return (
    <>
      <li key={todo.id} className="list-group-item">
        {editDesc ? (
          <form
            action=""
            className="formulario-editar-tarea"
            // onSubmit={handleSubmit}
          >
            <input
              type="text"
              className="formulario-editar-tarea__input"
              value={newDesc}
              onChange={(e) => setNewDesc(e.target.value)}
            />
            <button
              type="button"
              className="formulario-editar-tarea__btn"
              onClick={(e) => {
                handleUpdate(todo.id, newDesc);
                setEditDesc(false);
              }}
            >
              Update
            </button>
          </form>
        ) : (
          <div>
            <p
              className={`${todo.done && "completed"} `}
              //todo.id 626262626 send to function
              onClick={() => handleToogle(todo.id)}
            >
              {i + 1} - {todo.desc}
            </p>
            <button
              type="button"
              className="btn btn-outline-primary"
              //todo.id 626262626 send to function
              onClick={(e) => {
                e.preventDefault();
                changeEditDesc(!editDesc);
              }}
            >
              Update
            </button>
            <button
              className="btn btn-danger"
              //todo.id 626262626 send to function
              onClick={() => handleDelete(todo.id)}
            >
              Delete
            </button>
          </div>
        )}
      </li>
    </>
  );
};

export default TodoListItem;
