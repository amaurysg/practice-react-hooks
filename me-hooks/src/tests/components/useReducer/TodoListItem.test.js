import { shallow } from "enzyme";
import TodoListItem from "../../../components/useReducer/TodoListItem";
import demoTodos from "../../fixtures/demoTodos";

describe("Test in TodoListItem", () => {
  const handleDelete = jest.fn();
  const handleToogle = jest.fn();
  const handleUpdate = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      i={0}
      handleDelete={handleDelete}
      handleToogle={handleToogle}
      handleUpdate={handleUpdate}
    />
  );

  test("Debe render correct ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de llamar la funcion handleDelete, ", () => {
    //jest.fn()
    //toHaveBeenCalledWith
    wrapper.find(".btn-danger").simulate("click");
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("Debe de llamar la funcion handleToogle ", () => {
    //jest.fn()
    //toHaveBeenCalledWith
    wrapper.find("p").simulate("click");
    expect(handleToogle).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("Debe mostrar text correct ", () => {
    //contenido del parrafo
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(`1 - ${demoTodos[0].desc}`);
  });

  test("Debe tener class completed ", () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow(<TodoListItem todo={todo} />);

    expect(wrapper.find("p").hasClass("completed")).toBe(true);
  });
});
