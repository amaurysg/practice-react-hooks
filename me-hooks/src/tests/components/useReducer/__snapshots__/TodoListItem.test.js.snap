// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in TodoListItem Debe render correct  1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="1"
  >
    <div>
      <p
        className="false "
        onClick={[Function]}
      >
        1
         - 
        React
      </p>
      <button
        className="btn btn-outline-primary"
        onClick={[Function]}
        type="button"
      >
        Update
      </button>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </li>
</Fragment>
`;
